core/src/services/s3/backend.rs

```diff
@@ -517,6 +518,60 @@ impl S3Builder {
 
         self
     }
+
+    /// a helper function to make it easier to find region
```

Comment:
This is a public function to user, we should:

- Add detailed explaination on the input value.
- Add detailed explaination on the return value, like `Some(v)`, `None`
- Add an example for this API.